### PR TITLE
Implemented the skills NPC_RELIEVE_ON and NPC_RELIEVE_OFF

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: part of https://github.com/rathena/rathena/issues/1525

* **Server Mode**: re

* **Description of Pull Request**: 

Implemented the skills NPC_RELIEVE_ON and NPC_RELIEVE_OFF as part of the episode 17.2

### CODE REVIEW

File: src/map/battle.cpp [hunk 1]
```diff
@@ -1839,6 +1839,13 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 		pc_overheat(*sd, (battle_get_weapon_element(d, src, bl, skill_id, skill_lv, EQI_HAND_R, false) == ELE_FIRE ? 3 : 1));
 	}
 
+	sc = status_get_sc(bl);
```
Comment (aleos89) [inline src/map/battle.cpp:1842]: I'm only viewing this on the app, but was this not defined prior?

File: src/map/status.cpp [hunk 2]
```diff
@@ -10403,6 +10403,12 @@ int status_change_start(struct block_list* src, struct block_list* bl,enum sc_ty
 			break;
 		default:
 			break;
+	case SC_RELIEVE_ON:
+		status_change_end(bl, SC_RELIEVE_OFF, INVALID_TIMER);
+		break;
+	case SC_RELIEVE_OFF:
+		status_change_end(bl, SC_RELIEVE_ON, INVALID_TIMER);
+		break;
```
Comment (aleos89) [inline src/map/status.cpp:10411]: Maybe move these to EndOnStart?

File: src/map/battle.cpp (hunk 1 above)
Comment (Atemo) [reply src/map/battle.cpp:1842]: It's assigned multiple times but the last assignment of sc is for src
`sc = status_get_sc(src);`
and we want it for bl

Comment (aleos89) [reply src/map/battle.cpp:1842]: Variable re-usage, the chef's kiss. Would it be better then for us to just move this check up inside the bl if-statement that ends around line 1735~? Or is it needed to be down here to be after some other check? If it's the latter, maybe add an extra comment stating it needs to be after X, Y, Z check or something so we will know for future reference!

Comment (Atemo) [reply src/map/battle.cpp:1842]: I'm not sure but for me putting it at the end of `battle_calc_damage` (like `damagetaken`) is a safe spot

**Pull request merged**